[CI] DiskThresholdDeciderIT testRestoreSnapshotAllocationDoesNotExceedWatermarkWithMultipleRestores failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #8042 / debian-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/qv2poh27z6t2y)
- [elasticsearch-periodic #7937 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/gamva3so2bdnw)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT.testRestoreSnapshotAllocationDoesNotExceedWatermarkWithMultipleRestores" -Dtests.seed=FB8C2C891778B0BF -Dtests.locale=ki-KE -Dtests.timezone=Africa/Accra -Druntime.java=24
```

**Applicable branches:**
8.18

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.cluster.routing.allocation.decider.DiskThresholdDeciderIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testRestoreSnapshotAllocationDoesNotExceedWatermarkWithMultipleRestores),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a collection with size <1>
     but: collection size was <0>
```

**Issue Reasons:**
- [8.18] 2 failures in test testRestoreSnapshotAllocationDoesNotExceedWatermarkWithMultipleRestores (0.4% fail rate in 504 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderIT.java
@@ -56,7 +56,6 @@ import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcke
 import static org.hamcrest.Matchers.contains;
 import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.Matchers.in;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.lessThanOrEqualTo;
@@ -242,7 +241,7 @@ public class DiskThresholdDeciderIT extends DiskUsageIntegTestCase {
                 clusterAdmin().prepareRestoreSnapshot(TEST_REQUEST_TIMEOUT, "repo", "snap")
                     .setWaitForCompletion(true)
                     .setRenamePattern(indexName)
-                    .setRenameReplacement(indexName + "-copy")
+                    .setRenameReplacement(copyIndexName)
                     .execute(ActionTestUtils.assertNoFailureListener(restoreSnapshotResponse -> {
                         final RestoreInfo restoreInfo = restoreSnapshotResponse.getRestoreInfo();
                         assertThat(restoreInfo.successfulShards(), is(snapshotInfo.totalShards()));
@@ -268,10 +267,19 @@ public class DiskThresholdDeciderIT extends DiskUsageIntegTestCase {
 
         // wait for all the shards to finish moving
         safeAwait(allShardsActiveListener);
-        ensureGreen(indexName, indexName + "-copy");
+        ensureGreen(indexName, copyIndexName);
 
         final var tinyNodeShardIds = getShardIds(dataNodeId, indexName);
-        assertThat(tinyNodeShardIds, hasSize(1));
+        final var tinyNodeShardIdsCopy = getShardIds(dataNodeId, copyIndexName);
+        assertThat(
+            "expected just one shard from one index on the tiny node, instead got "
+                + tinyNodeShardIds
+                + " from the original index and "
+                + tinyNodeShardIdsCopy
+                + " from the copy",
+            tinyNodeShardIds.size() + tinyNodeShardIdsCopy.size(),
+            is(1)
+        );
         assertThat(tinyNodeShardIds.iterator().next(), in(shardSizes.getShardIdsWithSizeSmallerOrEqual(usableSpace)));
     }
 

